source/Octopus.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -46,18 +46,27 @@ protected virtual void CheckSpaceResource(IHaveSpaceResource spaceResource)
                         throw new ResourceSpaceDoesNotMatchRepositorySpaceException(spaceResource, space);
                 },
                 whenSystemScoped: () => { },
-                whenUnspecifiedScope: () =>
+                whenUnspecifiedScope: async () =>
```

Comment:
This isn't quite right yet - the caller doesn't await the result of this, so the exception may be ignored in some cases.

Feel free to grab me to help you resolve this over zoom. I may be difficult to get a hold of next week - you may have more success getting help from someone else in QRF that is good at `async/await`.